### PR TITLE
Highlight macro expansions in editor

### PR DESCRIPTION
**Pull request by levjj**
Added a checkbox to the editor which allows highlighting of macro names and expansions.  Source Maps are used to map areas in the generated code back to the macro that generated it.

In order to be less obtrusive, the macro highlighting is only shown when the cursor in the editor is at the corresponding macro name.

highlighting

### CODE REVIEW

File: src/sweet.js [hunk 1]
```diff
@@ -208,16 +208,15 @@
...
-        }
-        return {
-            code: codegen.generate(ast, _.extend({
+            result.code = output.code;
+            result.sourceMap = output.map.toString();
+        } else {
+            result.code = codegen.generate(ast, _.extend({
                 comment: true
             }, options.escodegen))
         };
```
Comment (disnet) [inline src/sweet.js]: broke the build because jshint doesn't like this unnecessary `;` :)

**Comment by disnet** [comment]
If I try and use the class macro from the homepage some of syntax is not getting highlighted. Any idea why not?

edit_sweet_js

**Comment by levjj** [comment]
Thanks for testing this out!  There was a bug in the actual CodeMirror highlighter that skipped certain parts of the text.  Fortunately, I was able to fix this with a one-line change.

**Comment by disnet** [comment]
Thanks!

**Pull request merged**